src/ucp/core/ucp_am.c

```diff
@@ -425,6 +440,35 @@ ucs_status_t ucp_worker_set_am_recv_handler(ucp_worker_h worker,
     return status;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_am_copy_user_header_to_internal_buf(ucp_request_t *req)
+{
+    void *user_header;
+    
+    if (!(req->send.msg_proto.am.header_length) ||
```

Comment:
header_length == 0

Reply:
i'm not sure i'm following, you prefer this syntax:
if ((req->send.msg_proto.am.header_length == 0) ||
over this one:
if (!(req->send.msg_proto.am.header_length) ||
?

Reply:
yes, basically we preffer exact comparison vs boolean expression for `0`, `NULL`, etc.

Reply:
np